### PR TITLE
[OSAC-257] StudioWeb activities for Cryptography package

### PR DESCRIPTION
**Pull request by mateisimionescu**
Description
StudioWeb activities for Cryptography package
How to test
Open Activity package in Studio Web
Jira
[OSAC-257](https://uipath.atlassian.net/browse/OSAC-257)
Label:
24S

### CODE REVIEW

File: Activities/Cryptography/UiPath.Cryptography.Activities.Packaging/UiPath.Cryptography.Activities.Packaging.csproj [hunk 1]
```diff
@@ -28,6 +30,11 @@
   </Target>
 
   <Target Name="AddDlls">
+	<!-- PDBs for debug-->
+	<ItemGroup Condition="'$(Configuration)' == 'Debug'">
+      <BuildOutputInPackage Include="$(OutputPath)UiPath.Cryptography.pdb" />
```
Comment (alcxs) [inline Activities/Cryptography/UiPath.Cryptography.Activities.Packaging/UiPath.Cryptography.Activities.Packaging.csproj:35]: alignment fix

File: Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/DecryptFileViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            base.InitializeModel();
+            int propertyOrderIndex = 1;
+
+            Algorithm.IsPrincipal = true;
+            Algorithm.OrderIndex = propertyOrderIndex++;
+            Algorithm.Widget = new DefaultWidget { Type = ViewModelWidgetType.Dropdown };
+
+            InputFilePath.IsPrincipal = true;
+            InputFilePath.OrderIndex = propertyOrderIndex++;
+            InputFilePath.Widget = new DefaultWidget { Type = ViewModelWidgetType.Input };
```
Comment (alcxs) [inline Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/DecryptFileViewModel.cs]: remove Inputs to leave SW to decide what better editor to use. Similar for all InArgument<string> properties

File: Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/DecryptFileViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+            Key.IsPrincipal = true;
+            Key.OrderIndex = propertyOrderIndex++;
+            Key.Widget = new DefaultWidget { Type = ViewModelWidgetType.Input };
+
+            OutputFilePath.IsPrincipal = true;
+            OutputFilePath.OrderIndex = propertyOrderIndex++;
+            OutputFilePath.Widget = new DefaultWidget { Type = ViewModelWidgetType.Input };
```
Comment (alcxs) [inline Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/DecryptFileViewModel.cs]: remove Inputs to leave SW to decide what better editor to use. Similar for all InArgument properties

File: Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/DecryptFileViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+            ContinueOnError.Value = false;
+        }
+
+        protected override async ValueTask InitializeModelAsync()
+        {
+            await base.InitializeModelAsync();
+        }
+
+        protected override void InitializeRules()
+        {
```
Comment (alcxs) [inline Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/DecryptFileViewModel.cs]: remove if you do not have additional code for this method

File: Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/DecryptFileViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+            Overwrite.OrderIndex = propertyOrderIndex++;
+            Overwrite.Widget = new DefaultWidget { Type = ViewModelWidgetType.Toggle };
+
+            ContinueOnError.IsPrincipal = false;
+            ContinueOnError.OrderIndex = propertyOrderIndex++;
+            ContinueOnError.Widget = new DefaultWidget { Type = ViewModelWidgetType.NullableBoolean };
+            ContinueOnError.Value = false;
+        }
+
+        protected override async ValueTask InitializeModelAsync()
```
Comment (alcxs) [inline Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/DecryptFileViewModel.cs]: remove if you do not have additional code for this method

File: Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/DecryptTextViewModel.cs [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+            base.InitializeModel();
+            int propertyOrderIndex = 1;
+
+            Algorithm.IsPrincipal = true;
+            Algorithm.OrderIndex = propertyOrderIndex++;
+            Algorithm.Widget = new DefaultWidget { Type = ViewModelWidgetType.Dropdown };
+
+            Input.IsPrincipal = true;
+            Input.OrderIndex = propertyOrderIndex++;
+            Input.Widget = new DefaultWidget { Type = ViewModelWidgetType.Input };
```
Comment (alcxs) [inline Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/DecryptTextViewModel.cs]: remove Input. see previous messages

File: Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/DecryptTextViewModel.cs [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+            Key.IsPrincipal = true;
+            Key.OrderIndex = propertyOrderIndex++;
+            Key.Widget = new DefaultWidget { Type = ViewModelWidgetType.Input };
```
Comment (alcxs) [inline Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/DecryptTextViewModel.cs]: remove Input. see previous messages

File: Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/DecryptTextViewModel.cs [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+
+            Encoding.IsPrincipal = false;
+            Encoding.OrderIndex = propertyOrderIndex++;
+            Encoding.Widget = new DefaultWidget { Type = ViewModelWidgetType.Input };
```
Comment (alcxs) [inline Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/DecryptTextViewModel.cs]: remove Input. see previous messages

File: Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/DecryptTextViewModel.cs [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+            Result.OrderIndex = propertyOrderIndex++;
+            Result.Widget = new DefaultWidget { Type = ViewModelWidgetType.Input };
+
+            ContinueOnError.IsPrincipal = false;
+            ContinueOnError.OrderIndex = propertyOrderIndex++;
+            ContinueOnError.Widget = new DefaultWidget { Type = ViewModelWidgetType.NullableBoolean };
+            ContinueOnError.Value = false;
+        }
+
+        protected override async ValueTask InitializeModelAsync()
```
Comment (alcxs) [inline Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/DecryptTextViewModel.cs]: remove override method if no additional code

File: Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/DecryptTextViewModel.cs [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+        {
+            await base.InitializeModelAsync();
+        }
+
+        protected override void InitializeRules()
```
Comment (alcxs) [inline Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/DecryptTextViewModel.cs]: remove override method if no additional code

File: Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/EncryptFileViewModel.cs [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+            base.InitializeModel();
+            int propertyOrderIndex = 1;
+
+            Algorithm.IsPrincipal = true;
+            Algorithm.OrderIndex = propertyOrderIndex++;
+            Algorithm.Widget = new DefaultWidget { Type = ViewModelWidgetType.Dropdown };
+
+            InputFilePath.IsPrincipal = true;
+            InputFilePath.OrderIndex = propertyOrderIndex++;
+            InputFilePath.Widget = new DefaultWidget { Type = ViewModelWidgetType.Input };
```
Comment (alcxs) [inline Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/EncryptFileViewModel.cs]: remove Input, see previous comments

File: Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/EncryptFileViewModel.cs [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+            Overwrite.OrderIndex = propertyOrderIndex++;
+            Overwrite.Widget = new DefaultWidget { Type = ViewModelWidgetType.Toggle };
+
+            ContinueOnError.IsPrincipal = false;
+            ContinueOnError.OrderIndex = propertyOrderIndex++;
+            ContinueOnError.Widget = new DefaultWidget { Type = ViewModelWidgetType.NullableBoolean };
+            ContinueOnError.Value = false;
+        }
+
+        protected override async ValueTask InitializeModelAsync()
```
Comment (alcxs) [inline Activities/Cryptography/UiPath.Cryptography.Activities/NetCore/ViewModels/EncryptFileViewModel.cs]: see previous comments

**Review by alcxs — CHANGES_REQUESTED** [review]
see comments and apply same to all view models

File: Activities/Cryptography/UiPath.Cryptography.Activities.Packaging/UiPath.Cryptography.Activities.Packaging.csproj (hunk 1 above)
Comment (alcxs) [reply Activities/Cryptography/UiPath.Cryptography.Activities.Packaging/UiPath.Cryptography.Activities.Packaging.csproj:35]: still an issue

File: Activities/Cryptography/UiPath.Cryptography.Activities.Packaging/UiPath.Cryptography.Activities.Packaging.csproj [hunk 13]
```diff
@@ -2,9 +2,12 @@
   <Import Project="..\Cryptography.build.props" />
   <PropertyGroup>
     <TargetFrameworks>net461;net5.0-windows;net5.0</TargetFrameworks>
-    <OutputPath>$(ProjectDir)..\..\Output\Activities\Cryptography\</OutputPath>
-    <VersionSuffix></VersionSuffix>
+    <OutputPath>..\..\Output\Activities\Cryptography\</OutputPath>
     <!--leave empty for ci to replace with prelease tag-->
+	<VersionSuffix></VersionSuffix>
```
Comment (alcxs) [inline Activities/Cryptography/UiPath.Cryptography.Activities.Packaging/UiPath.Cryptography.Activities.Packaging.csproj]: align issues, as well

File: Activities/Cryptography/UiPath.Cryptography.Activities/Properties/UiPath.Cryptography.Activities.resx [hunk 14]
```diff
@@ -59,319 +59,815 @@
...
+  </resheader>
+  <resheader name="writer">
+    <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <data name="AlgorithmDisplayName" xml:space="preserve">
     <value>Algorithm</value>
     <comment>property name</comment>
   </data>
-	<data name="EncodingDisplayName" xml:space="preserve">
+  <data name="EncodingDisplayName" xml:space="preserve">
```
Comment (alcxs) [inline Activities/Cryptography/UiPath.Cryptography.Activities/Properties/UiPath.Cryptography.Activities.resx]: why are this still here, please remove all unused resources

File: Activities/Cryptography/UiPath.Cryptography.Activities/Properties/UiPath.Cryptography.Activities.resx [hunk 15]
```diff
@@ -59,319 +59,815 @@
...
   </data>
-	<data name="EncodingDescription" xml:space="preserve">
+  <data name="EncodingDescription" xml:space="preserve">
     <value>The encoding used to interpret the input string.</value>
   </data>
-	<data name="EncryptAlgorithmDescription" xml:space="preserve">
+  <data name="EncryptAlgorithmDescription" xml:space="preserve">
     <value>The encryption algorithm that will be used by this activity. Usage of deprecated algorithms is not recommended.</value>
   </data>
-	<data name="EncryptTextEncodingDescription" xml:space="preserve">
```
Comment (alcxs) [inline Activities/Cryptography/UiPath.Cryptography.Activities/Properties/UiPath.Cryptography.Activities.resx:134]: why are this still here, please remove all unused resources

**Review by alcxs — CHANGES_REQUESTED** [review]
please review comments and fix

**Review by alcxs — APPROVED** [review]

**Review by ovidiuiliescu — APPROVED** [review]

**Pull request merged**